src/consumer.rs

```diff
@@ -309,9 +316,10 @@ pub async fn from_substrait_rel(
                     }
                 })
                 .collect::<Result<Vec<_>>>()?;
-            let left_cols: Vec<&str> = pairs.iter().map(|(l, _)| l.as_str()).collect();
-            let right_cols: Vec<&str> = pairs.iter().map(|(_, r)| r.as_str()).collect();
-            left.join(right, join_type, &left_cols, &right_cols, None)
+            let left_cols: Vec<Column> = pairs.iter().map(|(l, _)| l.clone()).collect();
+            let right_cols: Vec<Column> = pairs.iter().map(|(_, r)| r.clone()).collect();
+            left.join(&right, join_type, (left_cols, right_cols), None)?
```

Comment:
Is the last parameter (`filter`) that's set to `None` the same field semantically as Substrait `JoinRel`'s  [post_join_filter](https://github.com/substrait-io/substrait/blob/main/proto/substrait/algebra.proto#L161)?

If not, please ignore the comment.

If so, should we check if the `JoinRel` has `post_join_filter`? And if so, parse it?

Reply:
Yes, I think those probably are the same concept. It would be good to implement this in a separate PR.